### ISSUE TITLE
config gets overwritten

### ISSUE THREAD

**Issue by jaceee** [tpwalke2/BlueMapSignMarkers#69]
Every time I restart my fabric server the default config replaces my config

**Comment by tpwalke2** [comment]
Which version of Minecraft are you using?

Would you be able to attach your config file and a log file that shows the server startup?

**Comment by elboletaire** [comment]
> Every time I restart my fabric server the default config replaces my config

Uh, yeah, I also noticed that. You have to stop the server, then do changes, then boot it up again. When shutting the server down the mod always rewrites the file.

**Comment by tpwalke2** [comment]
@jaceee, can you confirm that you are encountering similar behavior as @elboletaire where you edit the config while the server is running but then shutting down the server overwrites the changes?

**Comment by elboletaire** [comment]
> @jaceee, can you confirm that you are encountering similar behavior as @elboletaire where you edit the config while the server is running but then shutting down the server overwrites the changes?

I think it's expected, considering you're writing the config on shutdown: https://github.com/tpwalke2/BlueMapSignMarkers/blob/main/src/main/java/com/tpwalke2/bluemapsignmarkers/BlueMapSignMarkersMod.java#L34

The actual problem is not the settings being overwritten in that point, but the fact that they get overwritten with the data on memory, rather than trying to read the file in that point to see if there are any changes, so that makes the file always look as the server was first boot up.

If the mod had some way to reload its config on game, I'd do that just before shutting it down. If there's some way to add this as a callback/event of bluemap's reload function, it would be a good idea to set this reload logic to the bluemap one too.

**Comment by tpwalke2** [comment]
Yeah, I think not saving on shutdown would be the simplest approach.

**Comment by tpwalke2** [comment]
@jaceee, @elboletaire, I have published a release to modrinth that contains this update: https://modrinth.com/mod/bluemap-sign-markers/version/1.21-0.4.0.30

**Comment by elboletaire** [comment]
> @jaceee, @elboletaire, I have published a release to modrinth that contains this update: [modrinth.com/mod/bluemap-sign-markers/version/1.21-0.4.0.30](https://modrinth.com/mod/bluemap-sign-markers/version/1.21-0.4.0.30)

Will try it as soon as I get out of office!

**Comment by elboletaire** [comment]
> @jaceee, @elboletaire, I have published a release to modrinth that contains this update: [modrinth.com/mod/bluemap-sign-markers/version/1.21-0.4.0.30](https://modrinth.com/mod/bluemap-sign-markers/version/1.21-0.4.0.30)

Sadly, I can't try yet, since my server is stuck on version 1.20.6 due to mods that still haven’t received updates 😢
